src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -937,10 +944,26 @@ private void ParseOption(string name, string value)
                     break;
                 case "ssl":
                     _ssl = ParseBoolean(name, value);
+                    _hasSslBeenSet = true;
                     break;
                 case "sslverifycertificate":
                     _sslVerifyCertificate = ParseBoolean(name, value);
                     break;
+                case "tls":
+                    _tls = ParseBoolean(name, value);
+                    break;
+                case "tlsallowinvalidcertificates":
+                    _sslVerifyCertificate = !ParseBoolean(name, value);
+                    _hasTlsAllowInvalidCertificatesBeenSet = true;
+                    break;
+                case "tlsallowinvalidhostnames":
+                    _sslVerifyCertificate = !ParseBoolean(name, value);
+                    _hasTlsAllowInvalidHostnamesBeenSet = true;
+                    break;
+                case "tlsinsecure":
+                    _sslVerifyCertificate = !ParseBoolean(name, value);
```

Comment:
Based on Java implementation, `tlsinsecure` has the same logic as for example `tlsallowinvalidhostnames`. 
https://github.com/mongodb/mongo-java-driver/commit/35ff868762bbff615e4d38c472045a96870b459e#diff-3ed3ce9e640a01622556f9a203da683bR509